File: CPP/1.两数之和.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int,int> mp;
        if(nums.size() <2) return res;
        for(int i = 0; i < nums.size(); i++){
            if(mp.count(nums[i])){
                return {i,mp[nums[i]]};
            }
            mp[target - nums[i]] = i;
        }
        return {};
    }
};

File: Python/1.两数之和.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) < 2:
             return [0,0]
        dict ={}
        i = 0
        while(i < len(nums)):
            if nums[i] in dict.keys():
                return [i,dict[nums[i]]]
            dict[target - nums[i]] = i
            i = i + 1
        return [0,0]